common/pipeline_storage/src/main/scala/uk/ac/wellcome/pipeline_storage/Indexer.scala

```diff
@@ -68,25 +68,35 @@ trait Indexable[T] {
   *   became an issue we'd probably want to do something like:
   *
   *   def version(image: AugmentedImage) =
-  *     10 * (2*image.version + image.source.version)
+  *     versionMultiplier * (2*image.version + image.source.version)
   */
 object Indexable extends Logging {
 
+  // We have typically seen nSources maxing out at about 650,
+  // so this multiplier should allow room for that plus a bit more.
+  final val versionMultiplier = 1000
+
   implicit val imageIndexable: Indexable[AugmentedImage] =
     new Indexable[AugmentedImage] {
       def id(image: AugmentedImage) =
         image.id.canonicalId
 
       def version(image: AugmentedImage) =
-        10 * (image.version + image.source.version) +
+        versionMultiplier * (image.version + image.source.version) +
           (image.source match {
-            case SourceWorks(_, _, nMergedSources) if nMergedSources >= 10 =>
-              throw new RuntimeException(
-                s"Image ${image.id.canonicalId} has $nMergedSources >= 10 merged sources; versioning/ingest may be inconsistent")
-            case SourceWorks(_, Some(redirected), nMergedSources) =>
-              nMergedSources + 1
-            case SourceWorks(_, None, nMergedSources) => nMergedSources
+            case SourceWorks(_, _, nSources) if nSources >= versionMultiplier =>
+              warnOnTooManySources(image, nSources)
+              // Clamp the increment so as not to override higher versions
+              versionMultiplier - 1
+            case SourceWorks(_, Some(redirected), nSources) => nSources
+            case SourceWorks(_, None, nSources)             => nSources - 1
           })
```

Comment:
I wonder if this might be clearer if you broke out the bit after the second `+`?

As in:

```scala
def version(image: AugmentedImage) = {
  val sourceOffset = (image.source match { … })

  versionMultiplier * (image.version + image.source.version) + sourceOffset
```